Themes: split single-site JSX in DotCom and Jetpack JSXs

folletto opened the pull request:
Architectural change: from a single `single-site.jsx` that handled rendering both single-site view for WordPress·com sites and single-site view for Jetpack sites, to two files with clear functional separation.

In detail:

* No visual changes. 
* No functional changes.
* `single-site.jsx` now only retains the splitting logic (`isJetpack`)
* `single-site-wpcom.jsx` is exclusively for .com sites.
* `single-site-jetpack.jsx` is exclusively for Jetpack sites (and handles error messages too).
* `jetpack-referrer-message.jsx` split out an existing message, similarly to the other two.

More cleanup is probably due, but this is a first PR that exclusively addresses the splitting.

### To test

* Open My Sites → Themes
* Try a .com site, and make sure everything works from that page (search, info, customize, activate, ...)
* Try a Jetpack site, and make sure everything works from that page (search, info, customize, activate, ...)
* Try a Jetpack site with Jetpack before 3.7
* Try a Jetpack site with Jetpack Manage disabled

ockham commented on client/my-sites/themes/single-site-jetpack.jsx:
```diff
@@ -0,0 +1,60 @@
+	( props ) => {
+		const { siteId, analyticsPath, analyticsPageTitle } = props;
+		const sites = sitesFactory();
+		const site = sites.getSelectedSite();
```
As discussed, let's pass this down from `single-site.jsx` as a prop.

folletto commented:
Ok updated:

* Moved `site` propagation one level up (to make future rework easier too)
* Removed explicit `analytics*` props to keep more symmetry (cleanup will happen in a future rework)

ockham commented:
Looks good, and tests well! I've left one nomenclature related note. Feel free to :ship: after addressing that!

folletto commented on client/my-sites/themes/single-site.jsx:
```diff
@@ -9,74 +9,18 @@ import { localize } from 'i18n-calypso';
-import JetpackManageDisabledMessage from './jetpack-manage-disabled-message';
-import { connectOptions } from './theme-options';
+import SingleSiteThemeShowcase from './single-site-wpcom';
+import SingleSiteThemeShowcaseJetpack from './single-site-jetpack';
```
Yeah I wasn't convinced because "Wpcom" seemed a weird capitalization of what's already an acronym, but ok. :)
